api/v1alpha1/envoyproxy_types.go

```diff
@@ -86,8 +86,46 @@ type EnvoyProxySpec struct {
 	//
 	// +optional
 	Shutdown *ShutdownConfig `json:"shutdown,omitempty"`
+
+	// FilterOrder defines the order of filters in the Envoy proxy's HTTP filter chain.
+	//
+	// +optional
+	FilterOrder []FilterOrder `json:"filterOrder,omitempty"`
+}
+
+// FilterOrder defines the order of filters in the Envoy proxy's HTTP filter chain.
+// The order of filters is determined by the order field, where the filter with
+// the lowest order value is applied first.
+// If unspecified, the default order of filters is applied.
+// Default order of filters:
+// - envoy.filters.http.cors               0
+// - envoy.filters.http.ext_authz          100
+// - envoy.filters.http.basic_authn        200
+// - envoy.filters.http.oauth2             300
+// - envoy.filters.http.jwt_authn          400
+// - envoy.filters.http.fault              500
+// - envoy.filters.http.local_ratelimit    600
+// - envoy.filters.http.rate_limit         700
+type FilterOrder struct {
+	Filter EnvoyFilter `json:"filter"`
+	Order  uint32      `json:"order"`
```

Comment:
lets make this `int32` (negative) so we can add filters ahead of `cors` w/o moving it